tests/unit/base/Plugin_unittest.cc

```diff
@@ -0,0 +1,151 @@
+}
+
+TEST(Plugin, library_copy_constructor)
+{
+    Library library("./libplugin.so");
+    Library copy_library(library);
+    EXPECT_EQ(copy_library, library);
+}
+
+TEST(Plugin, library_assignment_operator)
+{
+    Library library("./libplugin.so");
+    Library copy_library = library;
+    EXPECT_EQ(copy_library, library);
+}
+
+TEST(Plugin, library_manifest_accessor_name)
+{
+    std::string name = Library::getManifestAccessorName();
+    EXPECT_EQ(name, "shogunManifest");
+}
+
+TEST(Plugin, manifest)
+{
+    Manifest manifest1("Mock library",
+    {
+        std::make_pair("mock_base_class", erase_type(
+            MetaClass<CSGObject>(erase_type(
+                std::function<Some<CSGObject>()>(
+                    []() -> Some<CSGObject>
+                    {
+                        return Some<CSGObject>(new MockBaseClass);
+                    }
+                    ))))),
+    });
+
+    Manifest manifest2("Mock library",
+    {
+        std::make_pair("mock_base_class", erase_type(
+            MetaClass<CSGObject>(erase_type(
+                std::function<Some<CSGObject>()>(
+                    []() -> Some<CSGObject>
+                    {
+                        return Some<CSGObject>(new MockBaseClass);
+                    }
+                    ))))),
+    });
+
+    Manifest manifest3("Mock library",
+    {
+        std::make_pair("mock_base_class", erase_type(
+            MetaClass<CSGObject>(erase_type(
+                std::function<Some<CSGObject>()>(
+                    []() -> Some<CSGObject>
+                    {
+                        return Some<CSGObject>(new CGaussianKernel);
+                    }
+                    ))))),
+    });
+
+    Manifest manifest4("Library",
+    {
+        std::make_pair("mock_base_class", erase_type(
+            MetaClass<CSGObject>(erase_type(
+                std::function<Some<CSGObject>()>(
+                    []() -> Some<CSGObject>
+                    {
+                        return Some<CSGObject>(new MockBaseClass);
+                    }
+                    ))))),
+    });
+
+    EXPECT_EQ(manifest1.description(), "Mock library");
+    EXPECT_EQ(manifest1, manifest2);
+    EXPECT_NE(manifest1, manifest3);
+    EXPECT_NE(manifest1, manifest4);
+}
+    
+TEST(Plugin, metaclass)
+{
+    MetaClass<CSGObject> meta_class(erase_type(
+        std::function<Some<CSGObject>()>(
+            []() -> Some<CSGObject>
+            {
+                return Some<CSGObject>(new MockBaseClass);
+            }
+        )));
+
+    MetaClass<CSGObject> another_meta_class(erase_type(
+        std::function<Some<CSGObject>()>(
+            []() -> Some<CSGObject>
+            {
+                return Some<CSGObject>(new MockBaseClass);
+            }
+        )));
+
+    EXPECT_EQ(another_meta_class, meta_class);
+
+    Some<CSGObject> some_obj(new MockBaseClass);
+    Some<CSGObject> obj = meta_class.instance();
+    // EXPECT_EQ(obj, some_obj);
```

Comment:
@lisitsyn  This doesn't compile due to absence of `operator==` in Some. Shall I remove this check?